apps/web/src/components/WorkEmailCard.tsx/WipeWorkEmailDialog.tsx

```diff
@@ -0,0 +1,119 @@
+import { useState } from "react";
+import { useIntl } from "react-intl";
+import { FormProvider, useForm } from "react-hook-form";
+import { useMutation } from "urql";
+
+import { Dialog, Button } from "@gc-digital-talent/ui";
+import { commonMessages } from "@gc-digital-talent/i18n";
+import { graphql } from "@gc-digital-talent/graphql";
+import { toast } from "@gc-digital-talent/toast";
+
+const WipeWorkEmail_Mutation = graphql(/* GraphQL */ `
+  mutation WipeWorkEmail($id: UUID!) {
+    wipeUserWorkEmail(id: $id) {
+      id
+    }
+  }
+`);
+
+interface WipeWorkEmailDialogProps {
+  id: string;
+  workEmail: string;
+}
+
+const WipeWorkEmailDialog = ({ id, workEmail }: WipeWorkEmailDialogProps) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const intl = useIntl();
+
+  const methods = useForm();
+  const { handleSubmit } = methods;
+
+  const dialogTitle = intl.formatMessage({
+    defaultMessage: "Remove this work email?",
+    id: "PTfLmk",
+    description: "Dialog header",
+  });
+  const handleError = () => {
+    toast.error(
+      intl.formatMessage({
+        defaultMessage: "Error: failed to delete",
+        id: "jyECqY",
+        description: "Error for failing to delete",
+      }),
+    );
+  };
+
+  const [{ fetching }, executeWipeWorkInformationMutation] = useMutation(
+    WipeWorkEmail_Mutation,
+  );
+  const wipeWorkInformation = async () => {
+    await executeWipeWorkInformationMutation({ id })
+      .then((result) => {
+        if (result.data?.wipeUserWorkEmail) {
+          toast.success(
+            intl.formatMessage({
+              defaultMessage: "Work email deleted",
+              id: "3bVNFv",
+              description: "Successfully cleared work email information",
+            }),
+          );
+        } else {
+          void handleError();
+        }
+      })
+      .catch(handleError);
+  };
+  const handleWipe = async () => {
+    await wipeWorkInformation().then(() => {
+      setIsOpen(false);
+    });
```

Comment:
Hmm, usually when there is an error we leave the dialog open for easy retries. I think this would lead to it closing on error. Though, I dont have that strong of a feeling. 

However, we could just use `.finally()` so we don't need two functions to handle a single promise. If we want to leave the dialog open on error, the `setIsOpen` could jsut be moved to the success case.

Reply:
Sure, why not. I relocated the close. https://github.com/GCTC-NTGC/gc-digital-talent/pull/15149/commits/c9be43224fe65ee18f336bb4d6935b431058b615

Renamed files too in  https://github.com/GCTC-NTGC/gc-digital-talent/pull/15149/commits/d22bdda71dfa2c5a8f0cb982e426accfaf2b3fc0